### DISCUSSION TITLE
Single inverter setup is only seeing one meter?

### DISCUSSION THREAD

**Discussion by jpravetz** [WillCodeForCats/solaredge-modbus-multi#340] [Q&A]
My physical installation has one physical inverter, a backup interface, and one battery (LG RESU10H Prime) and am only seeing the two devices listed below. Excess solar panel production goes back to the grid.

- `solaredge_i1` (_SE7600H-USSNBBL14_)
- `solaredge_m1` (_SE-RGMTR-1D-240C-B_)

For m1 I have these attributes:

- Device ID: _1_
- Model: _SE-RGMTR-1D-240C-B_
- Option: _Production_
- Parent device ID: _1_
- Sunspec device: _Three Phase Wye Meter_
- Sunspec did: _203_

I believe there should be an import/export meter and a battery device that show up as well. _Any ideas as to why I am not seeing more devices?_

I installed this integration months ago, and am only getting back to debugging it now. I am updated to `v2.3.3` of this integration.  Thanks for any help or explanation.

Update: These are [Known Issues](https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues)

**Comment by WillCodeForCats** [comment]
Did you turn on detect batteries?

**Reply by jpravetz** [reply]
Yes, it is turned on in the integration's config. I should have mentioned this. Auto detect meters and batteries are checked.

**Reply by WillCodeForCats** [reply]
You can upload your diagnostics file and I can confirm it, but most likely the issue is your inverter doesn't support battery data over modbus.

The long explanation is here: https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues

The short version is SolarEdge enables battery data support at random and some people have been able to convince them to turn it on and some have not.

**Reply by jpravetz** [reply]
I recall reading that thread about battery support and beating up Solaredge support about enabling it. They said they did, but I saw no difference. I should share my email thread with them. Should I not also see a 2nd meter? Or is this a question for Solaredge support as well?

[config_entry-solaredge_modbus_multi-4848c32ab2f3d070ad779583f2ee4267.json.txt](https://github.com/WillCodeForCats/solaredge-modbus-multi/files/11775891/config_entry-solaredge_modbus_multi-4848c32ab2f3d070ad779583f2ee4267.json.txt)

**Reply by WillCodeForCats** [reply]
The reason you don't see two meters is also a similar reason: SolarEdge doesn't split data for the second function of a dual-function meter over the modbus interface (one physical meter that is programmed to provide two metering functions). So for "Production + E/I" or "Production + consumption" dual function meters only the first function (production) is exposed to modbus.

**Reply by jpravetz** [reply]
Yes I just read that in your known issues page (which has been updated since I last read it). I will read those issue threads for more info.

**Reply by WillCodeForCats** [reply]
I would think they could update their modbus code to report dual function meters properly (since obviously they are doing that for their own monitoring platform), but my feeling is that modbus isn't a priority for them.

**Comment by jpravetz** [comment]
Summary: these are [known issues](https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues) that are already covered by the documentation for this integration.
